src/mpid/ch4/shm/include/shm.h

```diff
@@ -143,9 +143,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_SHM_mpi_imrecv(void *buf, MPI_Aint count, MPI
                                                   MPIR_Request ** rreqp) MPL_STATIC_INLINE_SUFFIX;
 MPL_STATIC_INLINE_PREFIX int MPIDI_SHM_mpi_cancel_recv(MPIR_Request *
                                                        rreq) MPL_STATIC_INLINE_SUFFIX;
-MPL_STATIC_INLINE_PREFIX void *MPIDI_SHM_mpi_alloc_mem(size_t size,
-                                                       MPIR_Info *
-                                                       info_ptr) MPL_STATIC_INLINE_SUFFIX;
+MPL_STATIC_INLINE_PREFIX void *MPIDI_SHM_mpi_alloc_mem(size_t size, MPIR_Info * info_ptr,
+                                                       MPL_memory_class class)
```

Comment:
Not sure I agree with this. Memory classes are an MPL thing. IMO the shm code can supply the class when actually doing the allocation. In any case, I don't think this commit is really needed in this series, is it?

Reply:
Not necessarily. If you’re allocating a shared memory buffer, the shm module doesn’t know why. You have to tell it. This neutrons the change on the netmod side, I just missed the change at the time and while adding all of the function headers I noticed that this was a problem too. I’d rather it go in now so it doesn’t get forgotten.

Reply:
Ah, I see that it's actually part of the device interface. And we are always supplying `MPL_MEM_RMA`? That doesn't seem right. 1. Those buffers can be used for pt2pt. 2. We should differentiate user buffers from internal MPICH buffers.

Reply:
I don’t think so. Did an RMA get in there somewhere? It should assign it whatever class gets passed in.

Reply:
https://github.com/pmodels/mpich/blob/master/src/mpi/rma/alloc_mem.c#L94

Reply:
Oh. I see what you’re saying. Yeah, that could be used for something else, though I don’t know that we can figure that out since the user is just allocating the memory and then using it however they want. I assume it’ll be used for RMA but perhaps it could be a separate class.

Reply:
Right. I would either propose making a new "MPL_MEM_USER" class, or just not tracking it at all. I.e. not even using `MPL_malloc` since it's the user's responsibility. At some point we (@gcongiu) will be integrating other types of allocators and those won't have the MPL tracing anyways.

Reply:
I added a patch to change the memory class in `MPI_Alloc_mem` to `MPL_MEM_USER`.